### PR TITLE
Cache: Warmup does not exit if an shortcode call has an exception

### LINKED ISSUE #1636: Bug: Cache warmup schlägt fehl
Ein statischer Aufruf einer nicht-statischen Methode ```execute``` von ```MapShortcode``` führt in der 2.9.4 Version in unserer Instanz zu einer Exception und der Aufruf bricht ab.

Auswirkung ist denke ich gering, das Caching subsequenter normaler Seiten-Aufrufe greift ja auch ohne erfolgreichen warmup-Aufruf.

Folgender Fehler: ```Uncaught error: non-static method commonsbooking\map\baseshortcode::execute() cannot be called statically in /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-content/plugins/commonsbooking/src/service/cache.php:367```

Vollständiger Trace:

```
Uncaught error: non-static method commonsbooking\map\baseshortcode::execute() cannot be called statically in /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-content/plugins/commonsbooking/src/service/cache.php:367 
stack trace: 
#0 /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-content/plugins/commonsbooking/src/service/cache.php(263): commonsbooking\plugin::runshortcodecalls(array) 
#1 /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-includes/class-wp-hook.php(324): commonsbooking\plugin::warmupcache('') 
#2 /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-includes/class-wp-hook.php(348): wp_hook->apply_filters('', array) 
#3 /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-includes/plugin.php(517): wp_hook->do_action(array) 
#4 /home/webpages/lima-city/dasallrad/wordpress_de-2023-05-23-9d15cd/wp-admin/admin-ajax.php(192): do_action('wp_ajax_cb_cach...') 
#5 {main} thrown
```

Hier die betreffende Zeile:

https://github.com/wielebenwir/commonsbooking/blob/e7636a305f0c9c5cfb5978385dab088aeb503314/src/Service/Cache.php#L433

### PR DESCRIPTION
**Pull request by datengraben**
Fixes #1636 

TODOs:
* [x] Quickfix: Dem MapShortcode eine statische Methode zu spendieren, welche in dem `runShortcodeCalls` aufgerufen werden kann.
* [x] Außerdem denke wäre es gut, die Methode `runShortcodeCalls` durch das Auftreten einer Exception nicht komplett abbrechen zu lassen, sondern die restlichen Einträge in `cbShortCodeFunctions` noch durchlaufen zu lassen. Also ein `try-catch` zu platzieren und etwaige Exception trotzdem zu loggen.

@hansmorb Der Aktuelle Stand, mitigiert bereits das Verhalten im Issue: Es werden alle Shortcodes ausgeführt. Aber die auftretende Exception ist noch nicht behoben.

### CODE REVIEW

**Comment by datengraben** [comment]
@hansmorb die Methode `execute` ist jetzt statisch und an den relevanten Stellen geändert.

**Comment by datengraben** [comment]
Ich habe verzichtet Tests für den Shortcode zu schreiben. Kann aber noch nachgeholt werden.

**Review by hansmorb — APPROVED** [review]
Vielen Dank für den Fix!
